### PR TITLE
docs: added customize import strategy doc

### PR DESCRIPTION
**Pull request by shahednasser**
- Added Customize Import Strategy documentation
- Rearranged links under "How-to Guides" in the sidebar to show nested categories at the top.

Closes DOCS-232 and DOCS-253

### CODE REVIEW

File: docs/content/advanced/backend/batch-jobs/customize-import.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  processJob(batchJobId: string): Promise<void> {
+    throw new Error('Method not implemented.')
+  }
+  buildTemplate(): Promise<string> {
+    throw new Error('Method not implemented.')
+  }
+}
+
+export default ImportStrategy
+```
```
Comment (srindom) [inline docs/content/advanced/backend/batch-jobs/customize-import.md:61]: think we should make it clear that there are more methods to implement than just these two

File: docs/content/advanced/backend/batch-jobs/customize-import.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+### 3. Set the batchType Property
+
+Every batch job strategy class must have the static property `batchType` defined. It determines the type of batch job this strategy handles.
+
+Since only one batch job strategy can handle a batch job type, you can overwrite Medusa’s default batch job strategies by using the same `batchType` value in your custom strategy.
+
+So, for example, to overwrite the product import strategy set the `batchType` property in your strategy to `product-import`:
+
+```tsx
+class ImportStrategy extends AbstractBatchJobStrategy {
```
Comment (srindom) [inline docs/content/advanced/backend/batch-jobs/customize-import.md:72]:
```suggestion
class MyImportStrategy extends AbstractBatchJobStrategy {
```

File: docs/content/advanced/backend/batch-jobs/customize-import.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+### 3. Set the batchType Property
+
+Every batch job strategy class must have the static property `batchType` defined. It determines the type of batch job this strategy handles.
+
+Since only one batch job strategy can handle a batch job type, you can overwrite Medusa’s default batch job strategies by using the same `batchType` value in your custom strategy.
+
+So, for example, to overwrite the product import strategy set the `batchType` property in your strategy to `product-import`:
+
+```tsx
```
Comment (srindom) [inline docs/content/advanced/backend/batch-jobs/customize-import.md:71]: nit: technically this isn't tsx but pure typescript

```suggestion
```typescript
```

**Review by srindom — APPROVED** [review]
LGTM - w. comments

File: docs/content/advanced/backend/batch-jobs/customize-import.md (hunk 3 above)
Comment (shahednasser) [reply docs/content/advanced/backend/batch-jobs/customize-import.md]: That's true, it's exported from Notion in that format. I added a vale rule to spot it in the future

**Pull request merged**